### PR TITLE
Prevent hang forever

### PR DESCRIPTION
**Pull request by firecow**

### CODE REVIEW

File: src/job.ts [hunk 1]
```diff
@@ -437,36 +443,35 @@ export class Job {
...
-            shell: "bash",
-            stdio: ["pipe", "pipe", "pipe"],
-            cwd: this.cwd,
-        });
-
-        cp.stdin.write("set -eo pipefail\n");
+        let cmd = "set -eo pipefail\n";
+        cmd += "exec 0< /dev/null\n";
 
         if (this.imageName) {
```
Comment (moberghammer) [inline src/job.ts:449]: Maybe introduce imageJob instead of all the if this.imageName

Comment (firecow) [reply src/job.ts:449]: You talking about a JobBase.ts, ShellJob.ts and DockerJob.ts pattern? @moberghammer

File: src/job.ts [hunk 2]
```diff
@@ -312,6 +312,21 @@ export class Job {
         return `--env SSH_AUTH_SOCK=${process.env.SSH_AUTH_SOCK} -v ${process.env.SSH_AUTH_SOCK}:${process.env.SSH_AUTH_SOCK}`;
     }
 
+    private generateScriptsCmd(scripts: string[]) {
```
[the remark this answers is not in the document]
Comment (firecow) [reply src/job.ts]: It has been renamed.

File: src/job.ts (hunk 1 above)
Comment (moberghammer) [reply src/job.ts:449]: Yer something like that so you only need a single if

Comment (firecow) [reply src/job.ts:449]: Yay, its a lot of work for this. But its a good idea. Won't do it in this PR.

**Review by moberghammer — APPROVED** [review]
LGTM

**Pull request merged**